python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -112,26 +152,272 @@ def __init__(
+        os.environ["OMP_NUM_THREADS"] = str(num_cpus)
+        os.environ["MKL_NUM_THREADS"] = str(num_cpus)
+        os.environ["OPENBLAS_NUM_THREADS"] = str(num_cpus)
+        os.environ["BLIS_NUM_THREADS"] = str(num_cpus)
+
+        parallelise_cv = self._set_estimator_cv_parallelism(num_cpus, has_gpus)
+
+        with parallel_backend("ray", n_jobs=num_cpus):
+            start_time = time()
+            self.estimator.fit(X_train, y_train, **self.fit_params)
+            fit_time = time() - start_time
+
+            with tune.checkpoint_dir(step=1) as checkpoint_dir:
+                with open(os.path.join(checkpoint_dir, MODEL_KEY), "wb") as f:
+                    cpickle.dump(self.estimator, f)
+
+            if self.label_column:
+                validation_set_scores = self._score_on_validation_sets(
+                    self.estimator, datasets
+                )
+                cv_scores = self._score_cv(
+                    self.estimator,
+                    X_train,
+                    y_train,
+                    groups,
+                    # if estimator has parallelism, use that. Otherwise,
+                    # parallelize CV
+                    n_jobs=1 if not parallelise_cv else num_cpus,
+                )
+            else:
+                validation_set_scores = {}
+                cv_scores = {}
+
+        results = {
+            **validation_set_scores,
+            **cv_scores,
+            "fit_time": fit_time,
+        }
+        tune.report(**results)
+
+    def as_trainable(self) -> Type[Trainable]:
+        trainable_cls = super().as_trainable()
+        scaling_config = self.scaling_config
+
+        class SklearnTrainable(trainable_cls):
+            @classmethod
+            def default_resource_request(cls, config):
+                updated_scaling_config = config.get("scaling_config", scaling_config)
+                updated_scaling_config = ScalingConfigDataClass(
+                    **updated_scaling_config
+                )
+                ScalingConfigDataClass.validate_config(
+                    config=updated_scaling_config,
+                    allowed_keys=["trainer_resources"],
+                    scaling_config_arg_name="scaling_config",
+                    caller_name=self.__class__.__name__,
+                )
+
+                head_resources = (
+                    updated_scaling_config.trainer_resources
+                    if updated_scaling_config.trainer_resources
+                    else {"CPU": 1}
+                )
+                return PlacementGroupFactory([head_resources])
+
+            def _postprocess_checkpoint(self, checkpoint_path: str):
+                preprocessor = self._merged_config.get("preprocessor", None)
+                if not checkpoint_path or preprocessor is None:
+                    return
+                with open(os.path.join(checkpoint_path, PREPROCESSOR_KEY), "wb") as f:
+                    cpickle.dump(preprocessor, f)
```

Comment:
Should this be a method of the base Trainer class btw?

Reply:
If we decide on it, then it will be a separate PR anyway, so let's not block on this